src/NHSD.BuyingCatalogue.SolutionLists.Persistence/Repositories/SolutionListRepository.cs

```diff
@@ -29,10 +30,12 @@ public sealed class SolutionListRepository : ISolutionListRepository
                    INNER JOIN dbo.Capability AS cap
                            ON cap.Id = sc.CapabilityId
               LEFT OUTER JOIN dbo.FrameworkSolutions AS fs
-                           ON sol.Id = fs.SolutionId
+                           ON sol.Id = fs.SolutionId 
+                            
              WHERE ps.[Name] = 'Published'
                AND ISNULL(fs.IsFoundation, 0) = COALESCE(@foundationOnly, fs.IsFoundation, 0)
-               AND ci.SupplierId = ISNULL(@supplierId, ci.SupplierId);";
+               AND ci.SupplierId = ISNULL(@supplierId, ci.SupplierId)
+               AND ISNULL(fs.FrameworkId, '') = COALESCE(@frameworkId, fs.FrameworkId, '');";
```

Comment:
`FrameworkId` has a `NOT NULL` constraint so we can omit the `ISNULL` here. On that basis, we can also simplify the `COALESCE` to an `ISNULL` so the condition will be similar to the line above for `SupplierId`, as below:

```sql
AND fs.FrameworkId = ISNULL(@frameworkId, fs.FrameworkId);
```

Reply:
In query there is `LEFT outer join` on `dbo.FrameworkSolutions` table, which is making `frameworkId`  column null in where condition. And in SQL `NULL=NULL evaluates to false`, beacuse of which some tests are failing.
That is why I have used `AND ISNULL(fs.FrameworkId, '') = COALESCE(@frameworkId, fs.FrameworkId, '')` same as `IsFoundation` column which is also `NOT NULL`

Reply:
Updated Solution query with `INNER JOIN` between `Solution` and `FrameworkSolutions` table